app/Http/Controllers/User/EmailValidationController.php

```diff
@@ -11,6 +11,10 @@ class EmailValidationController extends Controller
 {
     public function validateEmail(Request $request, User $user)
     {
+        if (!(auth()->check() && auth()->user()->id == $user->id)) {
```

Comment:
The first check can be done with a middleware on the controller's __construct instead